posthog/api/organization.py

```diff
@@ -0,0 +1,96 @@
+from typing import Any, Dict
+
+from django.db import transaction
+from django.db.models import QuerySet, query
+from django.shortcuts import get_object_or_404
+from rest_framework import (
+    exceptions,
+    mixins,
+    permissions,
+    request,
+    response,
+    serializers,
+    status,
+    viewsets,
+)
+
+from posthog.models import Organization, OrganizationMembership, Team
+from posthog.permissions import CREATE_METHODS, OrganizationAdminWritePermissions, OrganizationMemberPermissions
+
+
+class PremiumMultiorganizationPermissions(permissions.BasePermission):
+    """Require user to have all necessary premium features on their plan for create access to the endpoint."""
+
+    message = (
+        "You must upgrade your PostHog plan to be able to create and administrate multiple projects in an organization."
+    )
+
+    def has_permission(self, request: request.Request, view) -> bool:
+        if (
+            request.method in CREATE_METHODS
+            and not request.user.is_feature_available("multistructure")
+            and request.user.organization.teams.count() >= 1
+        ):
+            return False
+        return True
+
+
+class OrganizationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Organization
+        fields = [
+            "id",
+            "name",
+            "created_at",
+            "updated_at",
+        ]
+        read_only_fields = [
+            "id",
+            "created_at",
+            "updated_at",
+        ]
+
+    def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Organization:
+        serializers.raise_errors_on_nested_writes("create", self, validated_data)
+        request = self.context["request"]
+        with transaction.atomic():
```

Comment:
Q: How is an atomic transaction different from a normal transaction?

Reply:
All transactions are atomic when using Postgres with Django, but `transaction.atomic()` widens the scope of atomicity from a single `INSERT/SELECT` to all queries inside the Python context, meaning if e.g. `Team.objects.create` here fails for some reason, it will be rolled back, but also `Organization.objects.create` and `OrganizationMembership.objects.create` will be rolled back.

Reply:
In postgres, all INSERT/UPDATE/DELETE queries get rolled back if the wrapping transaction fails. I guess that `.atomic` in this case is just creating a normal transaction in pg at least.